docker-compose.yaml

```diff
@@ -0,0 +1,142 @@
+version: '3.7'
+
+# This docker-compose file brings up a full Observation Portal stack with Task Scheduler,
+# Task Worker, and Example Frontend. There is also an associated container of redis for
+# caching and task scheduling. You will want to modify the DB_* environment variables to
+# connect to your db backend, and modify the *_URL environment variables to point to the
+# location where you have the associated services running, like CONFIGDB.
+services:
+    redis:
+        image: redis:3.2
+        command: ["redis-server", "--appendonly", "yes"]
+        restart: always
+        healthcheck:
+            test: ["CMD", "redis-cli", "ping"]
+            interval: 10s
+            timeout: 5s
+            retries: 30
+        ports:
+            -  6373:6379
+        volumes:
+            -  ./data/redis:/data
+
+    nginx:
+        image: nginx:1.19.0-alpine
+        ports:
+            - 8000:80
+        volumes:
+            - ./nginx.conf:/etc/nginx/conf.d/default.conf
+            - static_volume:/static
+        depends_on:
+            - observation_portal
+
+    observation_portal:
+        image: observatorycontrolsystem/observation-portal:3.5.4
+        expose:
+            - "8001"
+        environment:
+            # Note that 172.17.0.1 works for localhost on linux, but for mac you will
+            # need to use `host.docker.internal` instead to point to localhost.
+            - DB_HOST=172.17.0.1
```

Comment:
Since we have to set the same values for these env vars in a few different places in this file, it might be better to read the values from environment variables or a .env config file https://docs.docker.com/compose/environment-variables/ I'm sure someone will make the mistake of updating the DB_USER or the image tag or something in one service but not the others, and then be confused for a bit.

Reply:
yeah, that seems like a good idea